src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const toArray = sourceString.split(';');
+  const newObject = {};
+
+  toArray.forEach((part) => {
+    if (part.trim() !== '') {
+      const [property, value] = part.split(':').map((remove) => remove.trim());
+
+      if (property && value) {
+        newObject[property] = value;
+      }
+    }
+  });
```

Comment:
You can try to use reduce. In this case, you don't need initialize newObject on 10 line (**optional**)